### PR TITLE
Clean up singleton class eval statements

### PR DESCRIPTION
**Pull request by marcotc**
Quite a few of our `class_eval` and `class_eval + class << self` statements were due to not having access the previously private #prepend or #include methods.

This PR cleans up such usages.

Not all occurrences across the code base were converted, as some performed other operations (e.g. defining methods inside the `class << self` block). This PR touches the ones that are safe to do so, with minimal refactor.

### Ruby 2.0 deprecation checklist

- [x] Remove 2.0 from CI
- [x] Remove 2.0-specific code paths
- [ ] Update to 2.1+ standards: https://github.com/ruby/ruby/blob/v2_1_0/NEWS
  - [x] Module#include and Module#prepend are now public methods.
  - [x] Exception#cause _(no changes)_
  - [x] Array#to_h converts an array of key-value pairs into a Hash.
  - [x] Kernel#singleton_method _(no changes)_
  - [x] Process.clock_gettime
  - [x] Enumerable#to_h converts a list of key-value pairs into a Hash.
  - [x] **instance_eval/class_eval for accessing #include/#prepend**
  - [ ] Now the default values of keyword arguments can be omitted. Those “required keyword arguments” need giving explicitly at the call time.
- [ ] Update documentation

### CODE REVIEW

**Comment by ericmustin** [comment]
It looks like there's some issues with either the `exception_handler` changes itself for sucker punch, or the way we test exceptions

**Comment by marcotc** [comment]
I fixed the `sucker_punch` failures.

File: spec/ddtrace/contrib/rails/support/rails3.rb [hunk 1]
```diff
@@ -13,15 +13,13 @@
...
-    def inherited(base)
-      # raise "You cannot have more than one Rails::Application" if Rails.application
-      super
-      Rails.application = base.instance
-      Rails.application.add_lib_to_load_path!
-      ActiveSupport.run_load_hooks(:before_configuration, base.instance)
-    end
+Rails::Application.singleton_class.class_eval do
+  def inherited(base)
+    # raise "You cannot have more than one Rails::Application" if Rails.application
```
Comment (ivoanjo) [inline spec/ddtrace/contrib/rails/support/rails3.rb:18]: Unrelated to this change, but this commented out line is a bit suspect 🤔 .

**Review by ivoanjo — APPROVED** [review]
LGTM 👍

**Review by ericmustin — APPROVED** [review]
🚀 , 👋 2.0️⃣

File: spec/ddtrace/contrib/rails/support/rails3.rb (hunk 1 above)
Comment (marcotc) [reply spec/ddtrace/contrib/rails/support/rails3.rb:18]: ducks

**Pull request merged**